spec/requests/follow_relationships_spec.rb

```diff
@@ -0,0 +1,226 @@
+            delete follow_relationship_path(id: new_follow_relationship.id)
+          end.to change(FollowRelationship, :count).by(-1)
+        end
+
+        it 'deletes the following for the follower user' do
+          expect do
+            delete follow_relationship_path(id: new_follow_relationship.id)
+          end.to change { new_follow_relationship.follower.following.count }.by(-1)
+        end
+
+        it 'deletes the follower for the followed user' do
+          expect do
+            delete follow_relationship_path(id: new_follow_relationship.id)
+          end.to change { new_follow_relationship.followed.followers.count }.by(-1)
+        end
+
+        it 'returns 302 Found' do
+          delete follow_relationship_path(id: new_follow_relationship.id)
+          expect(response).to have_http_status(:found)
+        end
+
+        it 'redirects to the followed user profile' do
+          delete follow_relationship_path(id: new_follow_relationship.id)
+          should redirect_to(user_profile_path(new_follow_relationship.followed.nickname))
+        end
+
+        it 'returns an HTML response' do
+          delete follow_relationship_path(id: new_follow_relationship.id)
+          expect(response.content_type).to eq 'text/html; charset=utf-8'
+        end
+      end
+
+      context 'when the request format is JavaScript' do
+        let!(:new_follow_relationship) { create(:follow_relationship, follower: new_user) }
+
+        it 'deletes a follow relationship' do
+          expect do
+            delete follow_relationship_path(id: new_follow_relationship.id), xhr: true
+          end.to change(FollowRelationship, :count).by(-1)
+        end
+
+        it 'deletes the following for the follower user' do
+          expect do
+            delete follow_relationship_path(id: new_follow_relationship.id), xhr: true
+          end.to change { new_follow_relationship.follower.following.count }.by(-1)
+        end
+
+        it 'deletes the follower for the followed user' do
+          expect do
+            delete follow_relationship_path(id: new_follow_relationship.id), xhr: true
+          end.to change { new_follow_relationship.followed.followers.count }.by(-1)
+        end
+
+        it 'responds with success' do
+          delete follow_relationship_path(id: new_follow_relationship.id), xhr: true
+          expect(response).to have_http_status(:success)
+        end
+
+        it 'returns a JavaScript response' do
+          delete follow_relationship_path(id: new_follow_relationship.id), xhr: true
+          expect(response.content_type).to eq('text/javascript; charset=utf-8')
+        end
+
+        it 'changes the unfollow button to the follow button' do
+          delete follow_relationship_path(id: new_follow_relationship.id), xhr: true
+          expect(response.body).to include('Follow')
+        end
+      end
+
+      context 'when unfollowing a not followed user' do
+        let(:follow_relationship_id) { rand(1..100) }
```

Comment:
Instead of using a random, let’s use 0 to guarantee that it doesn’t exist (if you get 1 you’ll be pretty unlucky!)

Reply:
I changed it to 0, but in this context I don't have any follow relationship created or not?

I also deleted the line "before { delete follow_relationship_path(id: follow_relationship_id), xhr: true }" because it doesn't make sense to have it here (it just causes the relationship to be deleted twice)